### ISSUE TITLE
Robot Framework Page Object Package Up For Adoption

### ISSUE THREAD

**Issue by kahunacohen** [ncbi/robotframework-pageobjects#61]
Though support was never guaranteed, NCBI is no longer able to support this project. We've moved off of Robot Framework due to Robot Framework's lack of timely Python3 support and the fact that the majority of our developers/test-writers prefer to write tests in Python. You are free to either fork the project to address any issues or to adopt the project. Please comment on this issue if you are willing to adopt this project and maintain it going forward.  Thanks for everything, - NCBI.

**Comment by zephraph** [comment]
Could you add me as a collaborator? I can at least keep up with maintenance for now and perhaps roll some of these ideas into the S2L library.

**Comment by kahunacohen** [comment]
I can't add you as a collaborator because of the way organizational permissions work. You'd then have access to all our repos. I think I could move it over to your account and you could add collaborators...

**Comment by kahunacohen** [comment]
I could also move it over to https://github.com/robotframework

**Comment by zephraph** [comment]
That would work. You may want to ask pekka first.

**Comment by kahunacohen** [comment]
Can you please ask on my behalf? I honestly don't have much time left to
spend too much time on this, but would be glad to move the repo over.

**Comment by zephraph** [comment]
I pinged him. I'll leave the decision up to him. 

Ultimately I'd only have time for bug fixes, not new features. Still, if @pekkaklarck moves it, I'll at least do that much. Otherwise I'm not sure that I want to adopt it under my username because I can't promise full support or responsiveness.

**Comment by kahunacohen** [comment]
OK, whatever you guys want. As is, though, we will not really be able to
support anything beyond our immediate needs related to our existing tests,
sorry to say.

**Comment by zephraph** [comment]
That's fine, we understand. If not moved will the repository be deleted? If left alone someone will probably fork it at some point.

**Comment by kahunacohen** [comment]
No of course not

**Comment by kahunacohen** [comment]
So have we decided what we want to do with this repo?

**Comment by pekkaklarck** [comment]
First of all, thanks for creating this cool library. I haven't used it personally, but I have heard from others that it works pretty well. Thanks also for your valuable contributions to Robot Framework itself. It's unfortunate that you won't be using Robot Framework in the future, but I understand it is easier for developers to do all their testing in code. Python 3 support is something that will be fixed soon and, as you probably know, there is also [Python 3 compatible friendly fork](https://github.com/userzimmermann/robotframework) available. Anyway, if there ever is a need for non-programmers to create or understand tests again, you know where to find us. =)

The most important thing for the future of this library is finding a new maintainer. If it is OK for you, I'd like to keep this library here until that.

**Comment by tisto** [comment]
@pekkaklarck I started to look into Python 3 compatibility for this library. With this pull requests tests pass locally for me (https://github.com/ncbi/robotframework-pageobjects/pull/64). Though, before I move on I would like to set up a travis job that runs tests against Python 3. It seems there is a travis conf file but travis is not enabled. 

Moving the repo to the robotframework organisation would make it easier to fix this I guess...I could of course do the travis setup on my fork, but that would require to duplicate the work if someone else takes over at a later point.

**Comment by pekkaklarck** [comment]
@tisto, it's great that you are interested in this library. The main problem with it's Python 3 support is probably that Selenium2Library isn't yet Python 3 compatible, but I'm literally just now sending an email about addressing that issue to robotframework-users mailing list.

Going forward, the main issue with this library is that it needs a maintainer. Are you possibly interested? When to move external projects under the robotframework organization is something we probably should talk in general with a wider audience. So far the idea has been moving only very mature projects there, but I'm not sure is that a good approach.

Have you, by the way, noticed that @boakley [just released](https://groups.google.com/d/msg/robotframework-users/qqALnUQJUSA/o23Zhs_fEQAJ) a new, more light weight, page object library for Robot Framework?

**Comment by tisto** [comment]
@pekkaklarck Python 3 support for Selenium2Library is actually not much of a problem. I'm using https://github.com/HelioGuilherme66/robotframework-selenium2library/releases for robotframework-djangolibrary without any issues so far.

I started a very quick-and-dirty migration to Python 3 here (just to see how much work the actual porting would be):

https://github.com/ncbi/robotframework-pageobjects/pull/65/files

Thanks for the hint regarding the new lib. Will have a look before I proceed!

**Comment by pekkaklarck** [comment]
Yeah, making S2L itself Python 3 compatible isn't a big task. The PR by @HelioGuilherme66 already does that, but there were some issues that have prevented merging it. As I already wrote, [I'm trying to get some new life into the S2L project](https://groups.google.com/d/msg/robotframework-users/mml-Ji_nfmI/7_AOjZxbEwAJ) and will review the PR again once we get started.

It will be interesting to hear about your comments about these two page object libraries once you have had a change to look at the new one too. Pleas share your findings publicly on robotframework-users or elsewhere!

**Comment by frishberg** [comment]
@tisto, if you are interested in maintaining the library, let us know. As for Travis, I am not seeing that it is enabled either. @hellmanj, didn't we set it up at some point?

**Comment by hellmanj** [comment]
Yep, not sure what happened to it.

**Comment by conversica-aaronpa** [comment]
@frishberg I'd like to adopt this repo if I could, but not quite presently in position to do so, may work on that. Meanwhile curious about why no obvious generic Grid hook. Sauce implementation looks like it has most of what is needed, although Grid environmental complexity is an issue. Was such a change/addition ever done? TIA.

**Comment by davidk1** [comment]
@tisto, have you please published results from comparing the two page object libraries as  pekkaklarck asked you some time ago? See below:

**From pekkaklarck**: It will be interesting to hear about your comments about these two page object libraries once you have had a change to look at the new one too. Pleas share your findings publicly on robotframework-users or elsewhere!

Thank you.
